### PR TITLE
[Accessibility]: Announce success message after refreshing sidepane completes

### PR DESCRIPTION
**Pull request by flynnolivia**
In this PR:
- Trigger a notification at the end of `CDPTargetProviders.getChildren()` and hide it quickly to make the screen reader announce that the targets list has refreshed
- Updated tests

I'm not sure this is an ideal solution since it will also hide recent notifications that aren't related to our extension, but it seems to be what's available to fix the bug for now. There is a discussion at microsoft/vscode#114718 about exposing more API to control the screenreader which I can keep following.

### CODE REVIEW

File: src/extension.ts
Comment (mliao95) [inline src/extension.ts:71]: Have you thought about adding the notification to the end of `getChildren` so that it triggers on all refreshes?  Currently this only triggers if you press the refresh button and does nothing when destroying/creating new targets.

Comment (mliao95) [inline src/extension.ts:71]: It would be nice if it would say the number of targets found as well.  Maybe a message such as "Refresh complete.  {num} targets found"

Comment (flynnolivia) [reply src/extension.ts:71]: I wasn't sure if we want to announce it when creating/destroying targets because it will produce a lot more notifications and might get cluttered (I haven't seen a way to hide notifications after an amount of time yet, lmk if you know about that). Plus, if targets are created/destroyed by opening/closing tabs in the browser, the screen reader will already announce that the tab is being open/closed.

Putting it at the end of `getChildren` would be good though because it's eventually called anyway after `cdpTargetsProvider.refresh()` (the call to `getChildren` here is redundant but helps with getting a more accurate timing for the announcement). Also, I think there might be a similar accessibility issue with the close target button, that this would solve.

Comment (flynnolivia) [reply src/extension.ts:71]: Good idea, I'll add that!

File: src/cdpTargetsProvider.ts
Comment (mliao95) [inline src/cdpTargetsProvider.ts:83]: I think we should remove the `notifications.hideToasts` call unless we can specify our refresh toast.  We don't want to unexpectedly remove any other toast notifications for the user.

**Review by mliao95 — APPROVED** [review]
LGTM after the one change.  Thanks!

File: src/cdpTargetsProvider.ts
Comment (flynnolivia) [reply src/cdpTargetsProvider.ts:83]: Ok, I'll do that.
But I looked into this a little more and found a way to make only our toast close after a timeout using `vscode.window.withProgress`. I think makes a better UX for sighted users, the only thing is that NVDA doesn't announce the message, but Narrator does. With `showInformationMessage`, both NVDA and Narrator announce the message, but the toasts can stack up and the notification list can quickly get flooded with refresh messages. Which one should we go with?

Comment (vidorteg) [reply src/cdpTargetsProvider.ts:83]: I took a look at the discussion at:
https://github.com/microsoft/vscode/issues/114718

and IMO I think is better to wait for those fixes to land. (We could +1 the issue by mentioning it on our bug to improve visibility). I see value on this solution but I'm not sure it will fix the issue but just temporarily mitigate it, thoughts?

**Review by vidorteg — CHANGES_REQUESTED** [review]
left a comment

File: src/cdpTargetsProvider.ts
Comment (flynnolivia) [reply src/cdpTargetsProvider.ts:83]: I agree, I'll keep an eye on that discussion and follow up with this once the fixes land in vscode. Hopefully before July when all the accessibility bugs need to be fixed!

**Pull request closed without merging**
